### PR TITLE
Algolia sync

### PR DESCRIPTION
**Pull request by jimrandomh**
~Not tested; waiting on admin permissions to create some dev-prefixed indexes. Almost certainly breaks some things. Do not run pointed at a production instance without changing the index prefix first.~

This significantly refactors and fixes up how we deal with Algolia indexing. Where some fields cause things to be indexed or unindexed (for example, we don't index drafts), these are now rechecked, and if something should no longer be indexed, it's deleted. There are now functions for scanning the database and Algolia index for differences, and an optional cronjob to do this daily. There is no longer duplication between incremental- and batch-script Algolia indexing.

### CODE REVIEW

**Comment by jimrandomh** [comment]
Fixes #1620. Fixes #1580. Fixes #1388. Fixes #947. Fixes #728.

File: packages/lesswrong/server/search/utils.js [hunk 1]
```diff
@@ -95,9 +99,12 @@ Users.toAlgolia = (user) => {
   return [algoliaUser];
 }
 
-
 // TODO: Refactor this to no longer by this insane parallel code path, and instead just make a graphQL query and use all the relevant data
 Posts.toAlgolia = (post) => {
+  if (post.draft) return null;
```
Comment (Discordius) [inline packages/lesswrong/server/search/utils.js]: Just run `checkAccess` here, so we just have a consistent permission system

**Comment by jimrandomh** [comment]
Current overall state of Algolia indexing:
 * Incremental update and rescan fuctions mostly work.
 * The Algolia index config needs to make _id a searchable field on all indexes. We create these indexes manually through the Algolia admin UI, and this configuration needs to be done there, too.
 * Deleted comments stay indexed, because of a crash in Vulcan's `validateModifier` combined with a missing argument in `comments.moderate.async` and sloppy exception handling in Vulcan.
 * Code is inconsistent about whether a positive score is required to appear in search results.
 * Posts are split into multiple index entries on paragraph boundaries, so a search can't use terms from different paragraphs and still find them
 * Comments in the Algolia index are not split like posts are, and only the first 2000 characters of their body are searchable.

**Comment by jimrandomh** [comment]
Todos:
 * If an admin key is provided, update index settings
 * Resync on score change

File: packages/lesswrong/server/scripts/algoliaExport.js [hunk 2]
```diff
@@ -2,145 +2,127 @@
...
-    }
-  }
-  // eslint-disable-next-line no-console
-  console.log(`Exporting last ${importBatch.length} algolia objects`)
-  const err = await batchAdd(algoliaIndex, importBatch, false)
-  if (err) {
-    totalErrors.push(err)
-  }
+  
+  const totalErrors = [];
```
[the remark this answers is not in the document]
Comment (jimrandomh) [reply packages/lesswrong/server/scripts/algoliaExport.js]: This variable actually carried over from preexisting code where it was const. In any case, fixed.

**Pull request merged**